src/main/java/co/cask/hydrator/plugin/batch/aggreagtor/aggregator/Sampling.java

```diff
@@ -0,0 +1,238 @@
+            builder.set(field.getName(), record.get(field.getName()));
+        }
+        builder.set("random", random);
+        return builder.build();
+    }
+
+    /**
+     * Builds the schema for Reservoir sampling algorithm. Adding field for random value.
+     */
+    private Schema createSchemaWithRandomField(Schema inputSchema) {
+        List<Schema.Field> fields = new ArrayList<>();
+        fields.add(Schema.Field.of("random", Schema.of(Schema.Type.FLOAT)));
+        for (Schema.Field field : inputSchema.getFields()) {
+            fields.add(field);
+        }
+        return Schema.recordOf("schema", fields);
+    }
+
+    /**
+     * Config for Sampling Plugin.
+     */
+    public static class SamplingConfig extends PluginConfig {
+
+        @Nullable
+        @Description("The number of records that needs to be sampled from the input records.")
+        private Integer sampleSize;
+
+        @Nullable
+        @Description("The percenatage of records that needs to be sampled from the input records.")
+        private Float samplePercentage;
+
+        @Description("Type of the Sampling algorithm that needs to be used to sample the data.")
+        private String samplingType;
+
+        @Nullable
+        @Description("The percenatage of additional records that needs to be included in addition to the input " +
+                "sample size to account for oversampling to be used in Systematic Sampling.")
+        private Float overSamplingPercentage;
+
+        @Nullable
+        @Description("Random float value between 0 and 1 to be used in Systematic Sampling.")
+        private Float random;
+
+        @Nullable
+        @Description("Total number od input records.")
+        private Integer totalRecords;
+
+        public SamplingConfig() {
+            this.random = new Random().nextFloat();
+        }
+
+        public SamplingConfig(@Nullable Integer sampleSize, @Nullable Float samplePercentage,
+                              @Nullable Float overSamplingPercentage, @Nullable Float random,
+                              String samplingType, @Nullable Integer totalRecords) {
+            this.sampleSize = sampleSize;
+            this.samplePercentage = samplePercentage;
+            this.overSamplingPercentage = overSamplingPercentage;
+            this.random = random;
+            this.samplingType = samplingType;
+            this.totalRecords = totalRecords;
+        }
+
+        public void validate() {
+            if (sampleSize == null && samplePercentage == null) {
+                throw new IllegalArgumentException("Please provide Sample size or Sample Percentage values.");
+            }
+
+            if (samplingType.equalsIgnoreCase(TYPE.SYSTEMATIC.toString()) && totalRecords == null) {
+                throw new IllegalArgumentException("Please provide value for 'Total Records' when selecting sampling " +
+                        "type as 'Systematic'.");
+            }
```

Comment:
do you want to check for things like ranges here? like, if i enter a random number of 47?